### DISCUSSION TITLE
Automate the process

### DISCUSSION THREAD

**Discussion by Innoceana** [biigle/core#758] [Q&A] [answered]
Hi, 

I was wondering if somehow we can tell biigle which color is for a specific surface so the software or an add-inn can automate the creation of polygons based on the colors and textures we gave.
Thanks

**Comment by mzur** [comment]
Could you please elaborate? Do you want to automatically create labels (in a label tree) with specific colors? Or do you want to automatically create annotations (in images/videos)? Or something else?

**Reply by Innoceana** [reply]
Hi, thanks for the speedy reply.
I would like to create annotations. I am analyzing coverage of corals and entering every single annotation manually takes way too much time, I was wondering if we could tell the software which color/texture is which so it would create the annotations automatically. Does that make sense?
Thanks!

**Reply by mzur** [reply, marked as answer]
That's a lot you are asking for. In fact, this is what the entire field of computer vision is working on :wink: There is no easy solution.

In BIIGLE, we have a few computer vision/machine learning components that can assist during the annotation process. For example, there is MAIA that can automatically find interesting objects in images. Or there is Magic SAM that can automatically outline objects that you select. However, if I understand you correctly, we currently don't have a tool that can automatically select the correct label for an annotation (a point?) that you set.

If you do random/regular sampling for coverage analysis, you can also have a look at the [CoralNet](https://coralnet.ucsd.edu/) tool. This is a tool specialized for exactly this task and it also offers automatic processing to some degree.

In addition, I can say that we are working on a feature that may help or even automate such a task in BIIGLE, too. But it will still take some time (maybe a year or more) before it will be available.

**Reply by Innoceana** [reply]
Awesome! thanks for the update and related info. I appreciate it.
we will be tuned.

**Reply by mzur** [reply]
Be sure to subscribe to our [newsletter](https://biigle.de/newsletter) then you won't miss any new features :wink:
